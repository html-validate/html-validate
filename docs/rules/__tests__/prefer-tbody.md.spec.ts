import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<table>
	<tr><td>...</td></tr>
</table>`;
markup["correct"] = `<table>
	<tbody>
		<tr><td>...</td></tr>
	</tbody>
</table>`;

describe("docs/rules/prefer-tbody.md", () => {
	it("inline validation: incorrect", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"prefer-tbody":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"prefer-tbody":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
