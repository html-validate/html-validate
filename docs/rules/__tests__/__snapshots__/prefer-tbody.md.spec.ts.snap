// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/prefer-tbody.md inline validation: correct 1`] = `[]`;

exports[`docs/rules/prefer-tbody.md inline validation: incorrect 1`] = `
[
  {
    "errorCount": 1,
    "filePath": "inline",
    "messages": [
      {
        "column": 3,
        "line": 2,
        "message": "Prefer to wrap <tr> elements in <tbody>",
        "offset": 10,
        "ruleId": "prefer-tbody",
        "ruleUrl": "https://html-validate.org/rules/prefer-tbody.html",
        "selector": "table > tr",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<table>
	<tr><td>...</td></tr>
</table>",
    "warningCount": 0,
  },
]
`;
